File: Ruby/napakalaki/lib/monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

require_relative 'prize.rb'
require_relative 'bad_consequence.rb'

class Monster
  
  private_class_method :new
  
  def self.newStandard (name, combatLevel, prize, badConsequence)
    
    new(name, combatLevel, prize, badConsequence)
    
  end
  
  def self.newCultist(name, combatLevel, prize, badConsequence, levelChange)
    
    new(name, combatLevel, prize, badConsequence, levelChange)
    
  end
  
  def initialize(name, combatLevel, prize, badConsequence, levelChange = 0)
    @name = name
    @combatLevel = combatLevel
    @prize = prize
    @badConsequence = badConsequence
    @levelChangeAgainstCultistPlayer = levelChange
  end

  def getName
    @name
  end
  
  def getCombatLevel
    @combatLevel
  end
  
  def getPrize
    @prize
  end
  
  def getCombatLevelAgainstCultistPlayer
    @combatLevel + @levelChangeAgainstCultistPlayer
  end
  
  def getBadConsequence
    @badConsequence
  end
  
  def to_s
    "Monster name: #{@name} \nMonster level: #{@combatLevel} \nMonster level against Cultist: #{getCombatLevelAgainstCultistPlayer}\nPrize: #{@prize} \nBadconsequence: #{@badConsequence}"
  end
end
end

File: Java/src/napakalaki/Monster.java

package napakalaki;

public class Monster {
    
    private final String name;
    private final int combatLevel;
    private final Prize prize;
    private final BadConsequence badconsequence;
    private final int levelChangeAgainstCultistPlayer;

    /*
    A monster can be created with a levelchange against cultists or not.
    Then we have those two constructors.
    */
    
    public Monster(String name, int combatLevel, Prize prize, BadConsequence badconsequence) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.prize = prize;
        this.badconsequence = badconsequence;
        this.levelChangeAgainstCultistPlayer = 0;
    }
    
    public Monster(String name, int combatLevel, Prize prize, BadConsequence badconsequence, int levelChange) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.prize = prize;
        this.badconsequence = badconsequence;
        this.levelChangeAgainstCultistPlayer = levelChange;
    }

    public String getName() {
        return name;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return this.combatLevel + this.levelChangeAgainstCultistPlayer;
    }

    public int getCombatLevel() {
        return combatLevel;
    }

    public Prize getPrize() {
        return prize;
    }

    public BadConsequence getBadconsequence() {
        return badconsequence;
    }

    @Override
    public String toString() {
        return "\nName: " + name + "\nCombatLevel: " + combatLevel + "\nPrize: " + prize + "\nBadconsequence: " + badconsequence;
    }
    
    
}
